tests/tt_metal/tt_metal/llk/test_state_tracker.cpp

```diff
@@ -0,0 +1,161 @@
+            .set_page_size(out1_id, single_tile_size_bfp8b);
+    tt_metal::CreateCircularBuffer(program_, core, l1_output1_cb_config);
+
+    vector<uint32_t> compute_kernel_args = {};
+    std::map<std::string, std::string> defines;
+
+    defines["TT_METAL_STATE_TRACKER_TESTING_ENABLED"] = "1";  // Define to enable state tracker testing interface
+    defines["FORCE_WATCHER_OFF"] = "1";
+    defines["TT_METAL_LIGHTWEIGHT_KERNEL_ASSERTS"] = "1";
+
+    auto compute_kernel = tt_metal::CreateKernel(
+        program_,
+        "tests/tt_metal/tt_metal/test_kernels/compute/state_tracker.cpp",
+        core,
+        tt_metal::ComputeConfig{
+            .fp32_dest_acc_en = test_config.fp32_dest_acc_en,
+            .dst_full_sync_en = test_config.dst_full_sync_en,
+            .compile_args = compute_kernel_args,
+            .defines = defines});
+
+    SetRuntimeArgs(program_, compute_kernel, core, {});
+
+    distributed::EnqueueMeshWorkload(cq, workload, true);
+
+    // Always returns true. Failed lightweight asserts will hang the core, triggering triage timeout that dumps the
+    // call stack. For local testing, run ./tools/triage/dump_lightweight_asserts.py when a kernel hangs to print
+    // assert info and callstack.
+    return true;
+}
+}  // namespace unit_tests::compute::state_tracker
+
+////////////////////////////////////////////////////////////////////////////
+//                             Test Description
+// ------------------------------------------------------------------------
+// These tests aim to cover usage of these API calls:
+// - copy_tile_init
+// - copy_tile_to_dst_init_short
+// - copy_tile_to_dst_init_short_with_dt
+// - unpack_reconfig_data_format
+// - unpack_reconfig_data_format_srca
+// - unpack_reconfig_data_format_srcb
+// - pack_reconfig_l1_acc
+////////////////////////////////////////////////////////////////////////////
+
+// Test fixture that enables lightweight asserts for the duration of the test
+class MeshStateTrackerFixture : public MeshDeviceFixture {
+protected:
+    bool lightweight_prev_state{};
+    void SetUp() override {
+        lightweight_prev_state = tt::tt_metal::MetalContext::instance().rtoptions().get_lightweight_kernel_asserts();
+        tt::tt_metal::MetalContext::instance().rtoptions().set_lightweight_kernel_asserts(true);
+        MeshDeviceFixture::SetUp();
+    }
+
+    void TearDown() override {
+        tt::tt_metal::MetalContext::instance().rtoptions().set_lightweight_kernel_asserts(lightweight_prev_state);
+        MeshDeviceFixture::TearDown();
+    }
+};
+
+TEST_F(MeshStateTrackerFixture, TensixComputeStateTracker) {
+    unit_tests::compute::state_tracker::ReconfigConfig test_config = {
+        .num_tiles = 1, .fp32_dest_acc_en = false, .dst_full_sync_en = false};
+
+    for (unsigned int id = 0; id < devices_.size(); id++) {
+        ASSERT_TRUE(unit_tests::compute::state_tracker::single_core_state_tracker(devices_.at(id), test_config));
```

Comment:
```suggestion
        EXPECT_TRUE(unit_tests::compute::state_tracker::single_core_state_tracker(devices_.at(id), test_config));
```
It's a subtle difference. In case of ASSERT, if the call fails on first call (device 0), test will end. If we put EXPECT, test will still fail, but it will allow second iteration to execute (and all others after that).